### PR TITLE
refactor: software signing bitcoin

### LINKED ISSUE #4066: Refactor: separate tx generation from signing logic
Refactor required to extract ability to generate transaction without implicitly signing it

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5682759765).

Closes #4066

WIP

This PR takes on a refactor to fix the design problem where we implicitly sign bitcoin transactions as we generate them. This assumes key in memory, not the case with Ledger, where we need to build a separate user flow to sign the tx.

Open to ideas here as still considering different options. One would be to just not sign/finalise and invoke manually in each case. Another is to handle the split software/ledger logic internally in this hook.

### CODE REVIEW

File: src/app/common/transactions/bitcoin/use-generate-bitcoin-tx.ts [hunk 1]
```diff
@@ -17,22 +18,35 @@ interface GenerateNativeSegwitTxValues {
...
-      utxos: UtxoResponseItem[],
-      isSendingMax?: boolean
-    ) => {
-      if (!utxos.length) return;
-      if (!feeRate) return;
+  const { whenWallet } = useWalletType();
+
+  return useMemo(
+    () => ({
+      sign(tx: btc.Transaction) {
```
Comment (fbwoolf) [inline src/app/common/transactions/bitcoin/use-generate-bitcoin-tx.ts:29]: Would it make sense to separate this out into its own hook or rename the file to be more inclusive of both?

File: src/app/pages/send/send-crypto-asset-form/form/btc/use-btc-choose-fee.tsx [hunk 2]
```diff
@@ -47,21 +46,18 @@ export function useBtcChooseFee() {
...
-            recipient: txValues.recipient,
-            fee: feeValue,
-            feeRowValue,
-            time,
-          }),
-        ledger: noop,
-      })();
+
+      sign(txResp.tx);
+      sendFormNavigate.toConfirmAndSignBtcTransaction({
```
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/form/btc/use-btc-choose-fee.tsx:54]: I think this all looks good until I get here bc it appears signing is happening before routing to `toConfirmAndSignBtcTransaction`?

Comment (kyranjamie) [reply src/app/pages/send/send-crypto-asset-form/form/btc/use-btc-choose-fee.tsx:54]: Good point, this highlights how it already worked though. We're signing the transaction as soon as the user selects the fee, and then just broadcast on click at the end. We should probably not sign here at all....

**Comment by markmhendrickson** [comment]
@kyranjamie do you intend to complete this PR or hand over to @alter-eggo?

**Comment by kyranjamie** [comment]
Work done elsewhere

**Pull request closed without merging**